packages/vite/index.cjs

```diff
@@ -0,0 +1,36 @@
+/* eslint-disable no-restricted-globals */
+// Proxy ESM to be used in CJS
+
+module.exports.build = (...args) =>
+  import('./dist/node/index.js').then((i) => i.build(...args))
+
+module.exports.createServer = (...args) =>
+  import('./dist/node/index.js').then((i) => i.createServer(...args))
+
+module.exports.preview = (...args) =>
+  import('./dist/node/index.js').then((i) => i.preview(...args))
+
+module.exports.transformWithEsbuild = (...args) =>
+  import('./dist/node/index.js').then((i) => i.transformWithEsbuild(...args))
+
+module.exports.formatPostcssSourceMap = (...args) =>
+  import('./dist/node/index.js').then((i) => i.formatPostcssSourceMap(...args))
+
+module.exports.splitVendorChunkPlugin = (...args) =>
+  require('./dist/node-cjs/splitVendorChunk.cjs').splitVendorChunkPlugin(
+    ...args
+  )
```

Comment:
For async functions `build`, `createServer`, etc. we could use dynamic imports to redirect them to ESM while being compatible with CJS usage. But for sync functions like `mergeConfig`, `splitVendorChunkPlugin` here, we can't easily do that. Here I made a separate CJS build for `splitVendorChunkPlugin` as an example and inline poly-filled for `defineConfig` and `normalizePath`. If we really want, we could make CJS build for the rest as well (probably need some splitting to our codebase).

What are your opinions on this? 

- Try our best to provide CJS compactibility (might have duplications in our dist)
- Make public all functions async (that way we could use `import()` to redirect)
- Provide a warning stub for sync functions, and ask users to use ESM or dynamic import themselves. Like

```ts
module.exports.mergeConfig = () => throw new Error('"mergeConfig" is not avalible in CJS build of Vite. Use `const { mergeConfig } = await import('vite')` instead')
```

Reply:
Amazing! Maybe a mix of 1 And 3? Provide the most common ones, even if it requires duplication and push all other sync functions to be await imported by the user

Reply:
Done!